pkg/retry/retry.go

```diff
@@ -54,43 +61,46 @@ func WithBackoff(
 			return
```

Comment:
This comment addresses the `for` loop header line, but GitHub doesn't allow me to click there..

I would suggest incrementing `attempt`'s initial value to `1`.
- This would ease the attempt number usage and reduce off-by-ones, e.g., here:
  https://github.com/Icinga/icingadb/blob/2344e0a105e74236de00b8f863cdaa9227d12e29/pkg/icingadb/db.go#L686
- Or here:
  https://github.com/Icinga/icingadb/blob/2344e0a105e74236de00b8f863cdaa9227d12e29/pkg/icingadb/ha.go#L404-L406
- Furthermore, this would remove the directly firing `time.After` case:
  https://github.com/Icinga/icingadb/blob/b2fd89f7de91789d4bd806d54695ffea464796d3/pkg/retry/retry.go#L88
  https://github.com/Icinga/icingadb/blob/61f8e7f915c47b10036302d709b8254ef9e622c6/pkg/backoff/backoff.go#L37-L40

Reply:
Sounds good. I will do that.